libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/metrics/MapboxMetricsReporter.kt

```diff
@@ -0,0 +1,64 @@
+package com.mapbox.services.android.navigation.v5.internal.navigation.metrics
+
+import android.content.Context
+import com.google.gson.Gson
+import com.mapbox.android.telemetry.AppUserTurnstile
+import com.mapbox.android.telemetry.MapboxTelemetry
+import com.mapbox.navigation.utils.thread.WorkThreadHandler
+import com.mapbox.services.android.navigation.v5.utils.extensions.toTelemetryEvent
+
+object MapboxMetricsReporter : MetricsReporter {
+
+    private val gson = Gson()
+    private lateinit var mapboxTelemetry: MapboxTelemetry
+    @Volatile
+    private var metricsObserver: MetricsObserver? = null
+    private var threadWorker = WorkThreadHandler("MapboxMetricsReporter")
+
+    @JvmStatic
+    fun init(
+        context: Context,
+        accessToken: String,
+        userAgent: String
+    ) {
+        mapboxTelemetry = MapboxTelemetry(context, accessToken, userAgent)
+        mapboxTelemetry.enable()
+    }
+
+    // For test purposes only
+    internal fun init(
+        mapboxTelemetry: MapboxTelemetry,
+        threadWorker: WorkThreadHandler
+    ) {
+        this.mapboxTelemetry = mapboxTelemetry
+        this.threadWorker = threadWorker
+        mapboxTelemetry.enable()
+    }
+
+    @JvmStatic
+    fun disable() {
+        mapboxTelemetry.disable()
+    }
+
+    fun addAppTurnstileEvent(event: AppUserTurnstile) {
```

Comment:
We need, because AppTurnstile event is part of telemetry library, but not our SDK. So we cannot extend it with MetricEvent interface and we need to handle this event separately. I know it looks very ugly, but we cannot do anything with this right now(

Reply:
Have we considered wrapping it internally under a `MetricEvent`?

Reply:
I'm not sure, understand you clear... Could you clarify please what do you mean?